src/components/Calendar/Calendar.jsx

```diff
@@ -115,6 +115,22 @@ const Calendar = () => {
     month === 0 && year <= startingYear;
   const isDisabledNextYearButton = (month, year, endingYear) => month === 11 && year >= endingYear;
 
+  const setPeriodValues = (val) => {
+    if (selectedDate.length < 2) {
+      setSelectedDate([...selectedDate, val]);
+    } else {
+      setSelectedDate([val]);
+    }
+  };
+
+  const sortSelectedDates = () => {
+    if (selectedDate[1] && new Date(selectedDate[0]) > new Date(selectedDate[1])) {
+      selectedDate.reverse();
```

Comment:
`reverse` method mutates an array. And you can't do it. You need smthing like that:
```js
const arr = [...selectedDate];
arr.reverse()
setSelectedDate(arr);
```